python/tank/descriptor/io_descriptor/git.py

```diff
@@ -124,34 +127,27 @@ def _clone_then_execute_git_commands(self, target_path, commands):
         log.debug("Git clone into '%s' successful." % target_path)
 
         # clone worked ok! Now execute git commands on this repo
-        cwd = os.getcwd()
         output = None
-        try:
-            log.debug("Setting cwd to '%s'" % target_path)
-            os.chdir(target_path)
-            for command in commands:
-
-                full_command = "git %s" % command
-                log.debug("Executing '%s'" % full_command)
-
-                try:
-                    output = subprocess_check_output(
-                        full_command,
-                        shell=True
-                    )
-
-                    # note: it seems on windows, the result is sometimes wrapped in single quotes.
-                    output = output.strip().strip("'")
-
-                except SubprocessCalledProcessError as e:
-                    raise TankGitError(
-                        "Error executing git operation '%s': %s (Return code %s)" % (full_command, e.output, e.returncode)
-                    )
-                log.debug("Execution successful. stderr/stdout: '%s'" % output)
-
-        finally:
-            log.debug("Restoring cwd (to '%s')" % cwd)
-            os.chdir(cwd)
+        for command in commands:
+
+            # we use git -C to specify the working directory where to execute the command
+            full_command = "git -C \"%s\" %s" % (target_path, command)
```

Comment:
unc path fix: the approach with `os.chdir(cwd)` which didn't work for UNC paths has been replaced with a `git -C` style call which should be equivalent.